### PR TITLE
Implement #{ ... } syntax

### LINKED ISSUE #103: Cannot parse new dictionary literal `#{ ... }`

### PR DESCRIPTION
**Pull request by tyru**
Fix #103

### CODE REVIEW

File: autoload/vimlparser.vim [hunk 1]
```diff
@@ -3419,7 +3449,8 @@ function! s:ExprParser.parse_expr9()
         endif
       endwhile
     endif
-  elseif token.type == s:TOKEN_COPEN
+  elseif token.type == s:TOKEN_COPEN || token.type == s:TOKEN_RAWCOPEN
+    let is_rawdict = token.type == s:TOKEN_RAWCOPEN
```
Comment (k-takata) [inline autoload/vimlparser.vim]: `is_litdict` (literal dict) might be better?

File: autoload/vimlparser.vim [hunk 2]
```diff
@@ -2754,8 +2755,13 @@ function! s:ExprTokenizer.get2()
     call r.seek_cur(1)
     return self.token(s:TOKEN_COLON, ':', pos)
   elseif c ==# '#'
-    call r.seek_cur(1)
-    return self.token(s:TOKEN_SHARP, '#', pos)
+    if r.p(1) ==# '{'
+      call r.seek_cur(2)
+      return self.token(s:TOKEN_RAWCOPEN, '#{', pos)
+    else
+      call r.seek_cur(2)
```
Comment (mattn) [inline autoload/vimlparser.vim]: Why 2?

File: autoload/vimlparser.vim (hunk 1 above)
Comment (tyru) [reply autoload/vimlparser.vim]: litdict is short and seems reasonable. thanks!

File: autoload/vimlparser.vim (hunk 2 above)
Comment (tyru) [reply autoload/vimlparser.vim]: oops, will fix it. thanks!

**Comment by tyru** [comment]
@k-takata @mattn 
fixed!

File: autoload/vimlparser.vim [hunk 3]
```diff
@@ -2875,6 +2881,29 @@ function! s:ExprTokenizer.get_dstring()
   return s
 endfunction
 
+function! s:ExprTokenizer.get_dict_literal_key()
+  call self.reader.skip_white()
+  let r = self.reader
+  let c = r.peek()
+  if !s:isalnum(c) && c != '_' && c != '-'
+    throw s:Err(printf('unexpected token: %s', token.value), token.pos)
```
Comment (blueyed) [inline autoload/vimlparser.vim]: `token` is not defined here.

Comment (tyru) [reply autoload/vimlparser.vim]: oops, I missed the warning... (because current autoload/vimlparser.vim has lots of vint errors!)
fixed in #126

Comment (blueyed) [reply autoload/vimlparser.vim]: Thanks!

You might want to try this `.vintrc.yaml`:
```
policies:
  ProhibitNoAbortFunction:
    enabled: false
```

(FWIW Neomake includes the policy name in the output format)

**Pull request merged**